packages/app/src/utils/tenderSwapHooks.ts

```diff
@@ -59,6 +60,39 @@ export const useCalculateRemoveLiquidityOneToken = (
   return result?.value?.[0] ?? constants.Zero;
 };
 
+export const usePriceImpact = (
+  isSendingToken: boolean,
+  pool: string,
+  inputAmount: string,
+  receiveAmount: BigNumber
+) => {
+  const swapContract = new Contract(pool, TenderSwapABI) as TenderSwap;
+
+  const resultTenderToken = useCall(
+    pool && {
+      contract: swapContract,
+      method: "getToken0Balance",
+      args: [],
+    }
+  );
+  const resultUnderlyingToken = useCall(
+    pool && {
+      contract: swapContract,
+      method: "getToken1Balance",
+      args: [],
+    }
+  );
+  const tokenBalance = Number.parseInt(weiToEthWithDecimals(resultUnderlyingToken?.value?.[0] ?? constants.One, 5));
+  const tenderTokenBalance = Number.parseInt(weiToEthWithDecimals(resultTenderToken?.value?.[0] ?? constants.One, 5));
+
+  const spotPrice = isSendingToken ? tokenBalance / tenderTokenBalance : tenderTokenBalance / tokenBalance;
+  const spotOutputAmount = Number.parseInt(inputAmount ?? 0) / spotPrice;
+  const priceImpact =
+    (spotOutputAmount - Number.parseInt(weiToEthWithDecimals(receiveAmount ?? constants.One, 5))) / spotOutputAmount;
+
+  return { priceImpact };
+};
```

Comment:
I don't think we should first convert to decimals (`weiToEthwithDecimals`) and then parse an integer from it (`parseInt`). 

Because then we lose all the precision of the decimals. 

Instead it might be better to work with BigNumbers instead and only convert to decimal format at the end

Reply:
that what I was trying to do at the beginning, but for some reason the result after the `.div` was always `0x01`

Reply:
@vmaark ah yeah some versions of BigNumber don't do decimals, they expect integers only. 

We could convert to decimals using [`parseFloat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseFloat?retiredLocale=nl) instead of `parseInt`

Reply:
as long as we do just one division `float` should be fine I guess